config/routes.rb

```diff
@@ -1,3 +1,14 @@
-Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+Rails.application.routes.draw do
+  root 'foods#index'
+  devise_for :users
+  resources :foods
+  get 'public_recipes', to: "public_recipes#index", as: "public_recipes"
+  get 'general_shopping_list', to: "general_shopping_list#index", as: "general_shopping_list"
+  resources :recipes do
+    resources :recipe_foods, except: [:show]
+  end 
+  resources :foods, except: [:update, :show]
+  get 'users/index'
+  get 'users/show'

```

Comment:
- [ ] You have added two routes for `:foods` serving the same purpose. Check lines 4 and 10. I suggest you keep line 10 and delete line 4.